File: ios/Modules/Sources/Search/api/SearchRepository.swift
import Foundation
import DB

public protocol SearchRepository: Sendable {
    /// FTS-style query: prefix match with `*`, AND/OR, `"quoted phrase"`.
    func search(_ query: String, limit: Int) async -> [MessageRecord]
    func index(id: Int64, plaintext: String) async
    func unindex(_ id: Int64) async
}

File: android/app/src/main/java/sol/vortexx/android/search/api/SearchRepository.kt
package sol.vortexx.android.search.api

import sol.vortexx.android.db.entities.MessageEntity

interface SearchRepository {
    /**
     * Local full-text search over decrypted messages cached on this device.
     * FTS4 syntax: prefix match with `*`, boolean AND/OR, `"quoted phrase"`.
     */
    suspend fun search(query: String, limit: Int = 50): List<MessageEntity>

    /** Index a newly decrypted message. Called from ChatEngine. */
    suspend fun index(id: Long, plaintext: String)

    /** Remove from the index (on delete). */
    suspend fun unindex(id: Long)
}
